### ISSUE TITLE
Oxyplot not loading in PRISM WPF application

### ISSUE THREAD

**Issue by zwabbit** [oxyplot/oxyplot#537]
I have a line series plot in a WPF user control in a dll project that gets loaded by the main WPF application via unity. It's a pretty straightforward MVVM setup using Prism. The problem is when the program is run the oxyplot plot doesn't show up. I've tested to make sure the user control itself is loading, I replaced the plot with a built in textblock and that appeared. When viewing the user control in the designer the plot also shows up. It's only when the application is actually running that the plot does not appear. No exceptions are thrown and I don't see any errors so I don't know why the plot isn't being displayed. Below is the XAML for the user control:

```xaml
<UserControl x:Class="RamsauerModule.Views.RamsauerView"
             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
             xmlns:oxy="http://oxyplot.org/wpf" 
             mc:Ignorable="d" 
             d:DesignHeight="500" d:DesignWidth="900">
    <Grid>
        <Canvas Name="GraphCanvas" Background="White" Width="900" Height="300" VerticalAlignment="Top" HorizontalAlignment="Left">
            <Grid>
                <oxy:PlotView Name="GraphPlot" Title="Test" Margin="0,0,-900,-300">
                    <oxy:PlotView.Series>
                        <oxy:LineSeries ItemsSource="{Binding Points0}"/>
                    </oxy:PlotView.Series>
                </oxy:PlotView>
            </Grid>
        </Canvas>
        <Button Content="Initialize" HorizontalAlignment="Left" Margin="96,405,0,0" VerticalAlignment="Top" Width="75" Command="{Binding InitializeCommand}"/>
        <Button Content="Read Data" HorizontalAlignment="Left" Margin="208,405,0,0" VerticalAlignment="Top" Width="75" Command="{Binding MeasureCommand}"/>
        <Button Content="Save Data" HorizontalAlignment="Left" Margin="317,405,0,0" VerticalAlignment="Top" Width="75" Command="{Binding SaveCommand}"/>
        <Button Content="Clear Data" HorizontalAlignment="Left" Margin="436,405,0,0" VerticalAlignment="Top" Width="75" Command="{Binding ClearCommand}"/>
        <TextBox HorizontalAlignment="Left" Height="23" Margin="566,404,0,0" TextWrapping="Wrap" Text="{Binding FileName, UpdateSourceTrigger=PropertyChanged}" VerticalAlignment="Top" Width="120"/>
    </Grid>
</UserControl>
```

The DataContext is bound in the constructor and I know that much is working because the buttons trigger the appropriate method in the ViewModel class.

Version Info:
* Visual Studio 2013 Community Edition
* .NET 4.5
* PRISM 5.0
* Oxyplot 2014.1.546
* Unity 3.5

Basically for the nuget stuff whatever is the latest version uploaded to nuget.

**Comment by aateeque** [comment]
Can you share your ViewModel code? What is Points0? Does OxyPlot know how to plot Points0?

**Comment by zwabbit** [comment]
Points0 was just a List<DataPoint> instance. I'm trying to reproduce a slimmed down version of the issue but using the version of OxyPlot for WPF in nuget and the xmls of http://oxyplot.org/wpf, I can't seem to use the graph classes as specified on the http://docs.oxyplot.org/en/latest/getting-started/hello-wpf-xaml.html page. At present it seems like the code has outstripped the documentation to the point where I literally have no idea how I'm supposed to get things working again.

**Comment by aateeque** [comment]
That sounds like a dark scary place to be in mate!
Share some code and lets see what we can do ;)

**Comment by zwabbit** [comment]
Since github doesn't let me post zip files: https://drive.google.com/file/d/0B3rkcge3pwmbYVhHdjFScHNOaFE/view?usp=sharing

That's the current solution I'm trying to reproduce the issue in. Save for the fact that I can't seem to actually use the PlotView to create a plot anymore, none of the attributes used in the XAML example in the documentation is working anymore. The specific file where I was trying to insert a graph is in the PrismGraphModule Views\GraphView.xaml.

**Comment by objorke** [comment]
@zwabbit the docs were out of sync. Thanks for the notice! The [wpf/xaml page](http://docs.oxyplot.org/en/latest/getting-started/hello-wpf-xaml.html) is now updated. Use the `Plot` control when setting plot content in XAML. The example code referenced in the docs seems to be up to date.

**Comment by zwabbit** [comment]
Alright, here's a zip that demonstrates the specific problem:

https://drive.google.com/file/d/0B3rkcge3pwmbYWdVRk8wMi14d3c/view?usp=sharing

The viewmodel class is GraphViewModel and it doesn't "do" anything save initialize the list of DataPoints that then gets assigned to a property much as shown in the demo. To demonstrate that the data binding itself works there is a textblock that is bound to the property BlahText that is displaying.

**Comment by objorke** [comment]
Thanks for including a full solution!

**Comment by objorke** [comment]
This issue may be related to or duplicate of #40 #175 #630 #456

**Comment by jkoritzinsky** [comment]
I'm getting this same issue using the `<oxy:Plot>` tag. Here's my XAML:

```

                <oxyplot:Plot Title="{Binding Name}" LegendPlacement="Outside" LegendPosition="RightMiddle"
                            PlotType="XY">
                    <oxyplot:Plot.Axes>
                        <oxyplot:LinearAxis />
                        <oxyplot:LinearAxis />
                    </oxyplot:Plot.Axes>
                    <oxyplot:Plot.Series>
                        <oxyplot:LineSeries ItemsSource="{Binding OutputPoints}" />
                    </oxyplot:Plot.Series>
                </oxyplot:Plot>
```

**Comment by interdrift** [comment]
Just to confirm we are using PRISM & WPF, Oxyplot works fine.

**Comment by jkoritzinsky** [comment]
Just WPF. I'll try to make a minimal reproduction and I'll upload it. I can get the examples working, but not my code (which I've attempted to simplify even further after my post).

**Comment by tibel** [comment]
Is it working when you set `Width` and `Height` to the `Plot`?

**Comment by jkoritzinsky** [comment]
I tried that.  I finally got it to work, but only after having the `Height` set higher than the `Width`.  If they were too close, the renderer throws an exception and doesn't render it.  The exception was in `OxyRect` and it said that `height` was negative.

**Comment by jkoritzinsky** [comment]
Actually, it was just when the height was too small that it threw the exception.

**Comment by objorke** [comment]
The plot should handle such cases, not throw exceptions. Can you show some code that reproduces the issue?

https://github.com/oxyplot/oxyplot/blob/develop/Source/OxyPlot.Wpf/PlotBase.cs#L384
